src/main/java/org/apache/solr/benchmarks/beans/SolrBenchQueryResponse.java

```diff
@@ -0,0 +1,66 @@
+package org.apache.solr.benchmarks.beans;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.lang.invoke.MethodHandles;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.concurrent.atomic.AtomicReference;
+
+import org.apache.commons.io.IOUtils;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.solr.client.solrj.request.QueryRequest;
+import org.apache.solr.common.util.NamedList;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+// nocommit: javadocs
+public class SolrBenchQueryResponse {
+	
+    private static final Logger log = LoggerFactory.getLogger(MethodHandles.lookup().lookupClass());
+
+	public String queryString;
+	
+	public String responseString = null;
+	
+	public NamedList<Object> rawResponse;
+	
+	public boolean isSuccessfulResponse = false;
+	
+	public SolrBenchQueryResponse (String queryString, NamedList<Object> response) {
+		this.rawResponse = response;
+		this.queryString = queryString;
+		
+		InputStream responseStream = (InputStream) response.get("stream");
+		try {
+			responseString = getResponseStreamAsString(responseStream); // should only call this once, as this reads the stream!
```

Comment:
Is it possible to make reading the stream lazy? My concern is that this always read the response stream, which seems unnecessary if the caller does not care about detailed stats nor needing any validations (which i assume is a common case for general benchmarking?)